### DISCUSSION TITLE
Accessing this in the $watch magic method (As of alpinejs@3.4.2)

### DISCUSSION THREAD

**Discussion by ahmu83** [alpinejs/alpine#2271] [1. Help] [answered]
Accessing this in the $watch magic method (As of alpinejs@3.4.2)

Consider this component:

```
x-data="{
    open: false,

    init() {

      this.$watch('open', function(value) {

        this.toggleActiveClass()

      })

    },

    toggleActiveClass() {
      console.log('toggleActiveClass() function')
    },

  }"
```

This will result in "Uncaught TypeError: this.toggleActiveClass is not a function"

Should there be a magic method to access the main component object? i.e, $this?

**Comment by SimoTod** [comment, marked as answer]
Use an arrow function like the examples in the documentation.
`function() {...}` is different from `() => {}`, the former overrides `this`

**Reply by ahmu83** [reply]
Thank you, using an arrow function works.

I think there should be a magic method to access the main component object

**Reply by Tim-Wils** [reply]
`this.$root ` is the magic helper you’re looking for

**Reply by SimoTod** [reply]
I don't think it would fix your issue. Assuming a $this magic existed (returning the data context, $root returns an html element), you would call `this.$this` and if you don't use an arrow function, `this` won't work (also it doesn't make much sense).
